### ISSUE TITLE
Serializable attribute exists in both System.Runtime.Serialization.Formatters and nunit.framework

### ISSUE THREAD

**Issue by SerhiiS-in6k** [nunit/nunit#1751]
Hi I have issue with Serializable attribute
```
The type 'SerializableAttribute' exists in both 'System.Runtime.Serialization.Formatters, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' and 'nunit.framework, Version=3.4.1.0.
```
How can I specify the package to use? My project.json :
```
{
  "version": "1.0.0-*",
  "testRunner": "nunit",
  "buildOptions": {
    "debugType": "portable",
    "allowUnsafe": true,
    "define": [ "WIN64" ]
  },
  "frameworks": {
    "netcoreapp1.0": {
      "buildOptions": {
        "define": [ "NETSTANDARD1_6" ]
      },
      "dependencies": {
        "NETStandard.Library": "1.6.0",
        "System.Runtime": "4.1.0",
        "System.Runtime.Serialization.Formatters": "4.0.0-rc3-24212-01",
        "System.Dynamic.Runtime": "4.0.11",
        "System.Threading.ThreadPool": "4.0.10",
        "System.Reflection.TypeExtensions": "4.1.0",
        "System.Diagnostics.StackTrace": "4.0.1",
        "System.Diagnostics.Process": "4.1.0",
        "System.Threading.Thread": "4.0.0",
        "dotnet-test-nunit": "3.4.0-beta-1",
        "NUnit": "3.4.1"
      },
      "imports": "dnxcore50"
    },
  },
  "runtimes": {
    "ubuntu.14.04-x64": {},
    "win10-x64": {},
    "win7-x64": {}
  }
}

```

**Comment by rprouse** [comment]
Do you need to include the dependency on `System.Runtime.Serialization.Formatters`?

Longer term, the upcoming netstandard version of NUnit should likely use the `SerializableAttribute` in `System.Runtime.Serialization.Formatters` instead of redefining our own.

**Comment by SerhiiS-in6k** [comment]
Yes I need to include System.Runtime.Serialization.Formatters There few classes which I need.

**Comment by CharliePoole** [comment]
@rprouse Seems this is a bug but I'm not clear if it's an nunit bug or an agent bug. It seems to me we already removed this from the framework, but I haven't checked.

**Comment by rprouse** [comment]
It isn't defined in the Core agent, I think it is still the define in the Portable version of the framework, but I need to double check. Easy fix if it is will just be to give it a different namespace in portable.

**Comment by CharliePoole** [comment]
I removed it somewhere (can't remember but could search if it matters) by simply #if ing out the attribute where it doesn't exist. It seems like having it in the code could be deceptive to contributors reading the code, since the attributed class is actually __not__ serializable.

**Comment by SerhiiS-in6k** [comment]
Guys, Will you remove the attribute in a future release? Because I should support Net 4.0 framework too and I can't remove the attribute from project

**Comment by ChrisMaddock** [comment]
Is this a duplicate of #1683?

**Comment by CharliePoole** [comment]
It probably was a duplicate, but #1683 got fixed in a way that doesn't resolve the full problem. The fix to #1683 was to remove the use of SerializableAttribute by the nunitlite assembly, with no impact on nunit.framework.

Our compatible SerializableAttribute is only used in the Silverlight and Portable builds. It exists only for the purpose of saving a few lines by using #if. In fact, it's deceptive because the classes marked with our SerializableAttribute are not serializable!

Additionally, it's used in some places where it shouldn't be. I found a few classes that are both MBR and Serializable. They should be investigated as well and most likely have the attribute removed entirely.

My inclination is to change the places that use our Serializable attribute to

```C#
#if !SILVERLIGHT && !PORTABLE
[Serializable]
#endif

and then remove the definition.

**Comment by rprouse** [comment]
@CharliePoole I like that approach. It bothers me that we are stating classes are serializable when they are not.
